src/cli/command-processor.ts

```diff
@@ -0,0 +1,242 @@
+import {Worker} from 'worker_threads';
+import yargs from 'yargs/yargs';
+import * as readline from 'readline';
+
+/**
+ * An interface to formalize the message being sent to the worker thread.
+ */
+export interface IntentMessage {
+  intentType: string;
+  requestId: string;
+}
+
+/**
+ * A class containing all parameters needed to process an Identify command.
+ */
+export class IdentifyMessage implements IntentMessage {
+  intentType = 'IDENTIFY';
+  requestId: string;
+  discoveryBuffer: string;
+  deviceId: string;
+
+  /**
+   * @param requestId  The request id for a triggered Identify request.
+   * @param discoveryBuffer  The discovery buffer to send in a triggered Identify request.
+   * @param deviceId  The device id for a triggered Identify request.
+   * @returns  A new IntentMessage instance.
+   */
+  constructor(requestId: string, discoveryBuffer: string, deviceId: string) {
+    this.requestId = requestId;
+    this.discoveryBuffer = discoveryBuffer;
+    this.deviceId = deviceId;
+  }
+}
+
+/**
+ * A class containing all parameters needed to process an Identify command.
+ */
+export class ExecuteMessage implements IntentMessage {
+  intentType = 'EXECUTE';
+  requestId: string;
+  localDeviceId: string;
+  executeCommand: string;
+  params: Record<string, unknown>;
+  customData: Record<string, unknown>;
+
+  /**
+   * @param requestId  The request id for a triggered Execute request.
+   * @param localDeviceId  The localDeviceId for a triggered Execute request.
+   * @param executeCommand  The single command string to set in the triggered Execute request.
+   * @param params  The params array for the single Execute command.
+   * @param customData  The customData array for the single Execute command.
+   * @returns  A new ExecuteCommand instance.
+   */
+  constructor(
+    requestId: string,
+    localDeviceId: string,
+    executeCommand: string,
+    params: Record<string, unknown>,
+    customData: Record<string, unknown>
+  ) {
+    this.requestId = requestId;
+    this.localDeviceId = localDeviceId;
+    this.executeCommand = executeCommand;
+    this.params = params;
+    this.customData = customData;
+  }
+}
+
+export class CommandProcessor {
+  private worker: Worker;
+  constructor(worker: Worker) {
+    this.worker = worker;
+  }
+
+  // Wrapper for argument parsing with yargs.
+  private async parseIntent(userCommand: string): Promise<IntentMessage> {
+    const argv = await yargs()
```

Comment:
Not yet sure about all of these CLI parameter names

Reply:
I think splitting them into their separate commands will make these argument names less ambiguous.  Will do.

Reply:
Split into commands.